software/ADPL_electron/ADPL_electron.ino

```diff
@@ -10,13 +10,29 @@
  */
 
 SYSTEM_THREAD(ENABLED);  // parallel user and system threads
-                         // to allow function w/o cell connectivity
+// to allow function w/o cell connectivity
 
 unsigned long SYS_VERSION;
 
+#include "PublishDataCell.h"
+PublishDataCell cellPublisher;
+
 #define PUBLISH_DELAY 150000  // 2.5 min b/w variable publish
+#define sdIsInstalled true  // Determines existence of SD card - reevaluated in setup()
+
 
+#if sdIsInstalled
+#include "SD/SdFat.h"
+#include "PublishDataSD.h"
 #include "pin_mapping.h"
+// Software SPI.  Use any digital pins.
+// MISO => B1, MOSI => B2, SCK => B3, SS => B0
+SdFatSoftSpi<SD_DO_PIN, SD_DI_PIN, SD_CLK_PIN> sd;
+File sdFile;
+PublishDataSD sdPublisher;
+#else
+#include "pin_mapping.h"
```

Comment:
@CalNightingale it appears that `pin_mapping.h` is included regardless of the value of `sdIsInstalled`; can we move that outside the `if/else` structure, or is that required to get things to compile?  I would guess it should be able to be moved out... was this the error that you encountered before?